### ISSUE TITLE
Problems with differently templated user defined conversion operators

### ISSUE THREAD

**Issue by meator** [doxygen/doxygen#10184]
**Describe the bug**
Two different functions are treated as one.

**Screenshots**
![image](https://github.com/doxygen/doxygen/assets/67633081/889edee1-302d-49a6-aaa9-eab968cc62d7)

**To Reproduce**
Here is the code:
```c++
#include <vector>

/**
 * @brief Sample class convertible to a vector.
 */
template <typename T>
struct test
{
    operator const std::vector<T> &();

    template <typename U>
    operator const std::vector<U> &();
};

/**
 * @brief Convert test into vector.
 */
template <typename T>
test<T>::operator const std::vector<T> &()
{
    return std::vector<T>();
}

/**
 * @brief Convert test into a custom vector of type _U_.
 */
template <typename T>
template <typename U>
test<T>::operator const std::vector<U> &()
{
    return std::vector<U>();
}
```
The default Doxyfile can be used.

**Version**
`1.9.7`, Voidlinux. I am using my distro's doxygen.

**Additional context**
This is a pretty complicated situation. I am unsure if this is the intended behavior. If it is, is there a way to separate these (using `@` command or a option in Doxyfile)? It looks like Doxygen knows that there are two operators because the "Public Member Functions" section has two entries.

**Comment by doxygen** [comment]
@meator Please verify if the referenced commit fixes the issue for you. Do close the issue, this will be done automatically when the next official release is made available.

**Comment by meator** [comment]
I can confirm this fixes my problem. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
